src/Database/Query.php

```diff
@@ -1393,6 +1394,27 @@ public function into($table)
     }
 
     /**
+     * Creates an expression that refers to an identifier. Identifiers are used to refer to field names and allow
+     * the SQL compiler to apply quotes or escape the identifier.
+     *
+     * The value is used as is, and you might be required to use aliases or include the table reference in
+     * the identifier. Do not use this method to inject SQL methods or logical statements.
+     *
+     * ### Example
+     *
+     * ```
+     * $query->newExp()->lte('count', $query->identifier('total'));
+     * ```
+     *
+     * @param string $identifier The identifier for an expression
+     * @return ExpressionInterface
+     */
+    public function identifier($identifier)
+    {
+        return new IdentifierExpression($identifier);
```

Comment:
@thinkingmedia That alone unfortunately doesn't really address the incompatibility issue, which to me is the bigger problem.

An interface that isn't actually used in the core, cannot represent a realiable contract. From just looking at the API, it would seem that any object that implements that interface suffices any possible requirements in the core, but that isn't the case.